associated-token-account/program/src/processor.rs

```diff
@@ -118,3 +125,132 @@ pub fn process_create_associated_token_account(
         ],
     )
 }
+
+/// Processes `CloseNested` instruction
+pub fn process_close_nested(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let nested_associated_token_account_info = next_account_info(account_info_iter)?;
+    let nested_token_mint_info = next_account_info(account_info_iter)?;
+    let destination_token_account_info = next_account_info(account_info_iter)?;
+    let destination_system_account_info = next_account_info(account_info_iter)?;
+    let owner_associated_token_account_info = next_account_info(account_info_iter)?;
+    let owner_token_mint_info = next_account_info(account_info_iter)?;
+    let wallet_account_info = next_account_info(account_info_iter)?;
+    let spl_token_program_info = next_account_info(account_info_iter)?;
+    let spl_token_program_id = spl_token_program_info.key;
+
+    // Check owner address derivation
+    let (owner_associated_token_address, bump_seed) =
+        get_associated_token_address_and_bump_seed_internal(
+            wallet_account_info.key,
+            owner_token_mint_info.key,
+            program_id,
+            spl_token_program_id,
+        );
+    if owner_associated_token_address != *owner_associated_token_account_info.key {
+        msg!("Error: Owner associated address does not match seed derivation");
+        return Err(ProgramError::InvalidSeeds);
+    }
+
+    // Check nested address derivation
+    let nested_associated_token_address = get_associated_token_address_with_program_id(
```

Comment:
`with_program_id` refers to the token program id, and not the associated token account program id.  We don't need to call the "internal" version because we're not using the bump seed, but I could change the call to use `get_associated_token_address_and_bump_seed_internal` instead

Reply:
Oh, duh. I forgot where I was.  Thanks for the course correction!